### PR TITLE
Realtime communication

### PR DESCRIPTION
**Pull request by luisandia**

### CODE REVIEW

**Comment by luisandia** [comment]
Getting sha1-key:

keytool -list -v -keystore ~/.android/debug.keystore -alias androiddebugkey -storepass android -keypass android

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -52,9 +52,9 @@ limitations under the License.
       android:screenOrientation="landscape">
 
       <intent-filter>
-        <action android:name="android.intent.action.MAIN" />
+        <!--action android:name="android.intent.action.MAIN" />
```
Comment (alculquicondor) [inline app/src/main/AndroidManifest.xml]: remove these lines

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -64,6 +64,24 @@ limitations under the License.
...
 
+
+    <meta-data android:name="com.google.android.gms.games.APP_ID"
+        android:value="@string/app_id"/>
+    <meta-data android:name="com.google.android.gms.version"
+        android:value="@integer/google_play_services_version"/>
+
+    <activity
+        android:name=".Menu"
+        android:label="VRaze"
```
Comment (alculquicondor) [inline app/src/main/AndroidManifest.xml]: `@string/app_name`

File: app/src/main/cpp/vraze_jni.cc [hunk 3]
```diff
@@ -61,3 +61,8 @@ NATIVE_METHOD(void, nativeOnDestroy)
 (JNIEnv *env, jobject obj, jlong controller_paint_jptr) {
   delete ptr(controller_paint_jptr);
 }
+
+NATIVE_METHOD(void, nativeMoveCar)
+(JNIEnv* env, jobject obj, jlong controller_paint_jptr,float x, float y, float dir0, float dir1){
```
Comment (alculquicondor) [inline app/src/main/cpp/vraze_jni.cc]: `dir_x` and `dir_y`

File: app/src/main/java/pe/edu/ucsp/vraze/MainActivity.java [hunk 4]
```diff
@@ -63,6 +65,7 @@ public void run() {
   public static void messageMe(float x, float y, float dir0, float dir1) {
     Log.d(TAG,"x: " + String.valueOf(x) + " y: " + String.valueOf(y));
```
Comment (alculquicondor) [inline app/src/main/java/pe/edu/ucsp/vraze/MainActivity.java]: We won't be needing these logs anymore. But we can remove them later.

File: app/src/main/java/pe/edu/ucsp/vraze/MainActivity.java [hunk 5]
```diff
@@ -199,4 +202,22 @@ public void onDrawFrame(GL10 gl) {
...
+   */
+  public static RealTimeMessageReceivedListener mMessageReceivedListener = new RealTimeMessageReceivedListener() {
+    @Override
+    public void onRealTimeMessageReceived(RealTimeMessage rtm) {
+      byte[] buf = rtm.getMessageData();
+      String sender = rtm.getSenderParticipantId();
+      Log.d(TAG, "x" + String.valueOf(buf[0]));
+      Log.d(TAG, "y" + String.valueOf(buf[1]));
+      Log.d(TAG, "dir: " + (float) buf[2] + "/" + (float) buf[3]);
+
```
Comment (alculquicondor) [inline app/src/main/java/pe/edu/ucsp/vraze/MainActivity.java]: you can already call the `nativeMoveCar`

File: app/src/main/java/pe/edu/ucsp/vraze/Menu.java [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+
+  // Called when we get an invitation to play a game. We react by showing that to the user.
+
+
+  // Broadcast my score to everybody else.
+  public static void SendToPlayers(float x, float y, float dir0, float dir1, boolean is_finish) {
+    if (!mMultiplayer)
+      return; // playing single-player mode
+
+    mMsgBuf[0] = (byte) x;
```
Comment (alculquicondor) [inline app/src/main/java/pe/edu/ucsp/vraze/Menu.java]: https://stackoverflow.com/questions/14619653/converting-a-float-to-a-byte-array-and-vice-versa-in-java

**Review by alculquicondor — CHANGES_REQUESTED** [review]

**Comment by alculquicondor** [comment]
Looks good. Let's wait for the fix for the Daydream activity. Then we can cleanup code and merge.

**Pull request merged**